pyiceberg/catalog/hive.py

```diff
@@ -407,7 +407,30 @@ def register_table(self, identifier: Union[str, Identifier], metadata_location:
         raise NotImplementedError
 
     def list_views(self, namespace: Union[str, Identifier]) -> List[Identifier]:
-        raise NotImplementedError
+        """List Iceberg views under the given namespace in the catalog.
+
+        When the database doesn't exist, it will just return an empty list.
+
+        Args:
+            namespace: Database to list.
+
+        Returns:
+            List[Identifier]: List of view identifiers.
+
+        Raises:
+            NoSuchNamespaceError: If a namespace with the given name does not exist, or the identifier is invalid.
+        """
+        database_name = self.identifier_to_database(namespace, NoSuchNamespaceError)
+        with self._client as open_client:
+            return [
+                (database_name, view.tableName)
+                for view in open_client.get_table_objects_by_name(
+                    dbname=database_name, 
+                    tbl_names=open_client.get_all_tables(db_name=database_name)
+                )
+                if view.parameters.get(TABLE_TYPE, "").lower() == ICEBERG and view.tableType.lower() == "virtual_view"
```

Comment:
nit: can we do something like this here?
https://github.com/apache/iceberg-python/commit/d5b51f9eb14c6d8b8547b9bcdd3ec3059bdf3236#diff-5dc4045c0eac182d6c22d8a06f62fcc79c369ad8bff80d357252147e2d9a76f2R785
for both `__is_iceberg_table` and `__is_iceberg_view`